plugins/modules/content_export_repository.py

```diff
@@ -0,0 +1,136 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: content_export_repository
+version_added: 3.5.0
+short_description: Export repository content to a directory
+description:
+    - Export repository content to a directory.
+author:
+    - "Jeremy Lenz (@jeremylenz)"
+options:
+  repository:
+    description:
+      - Name of the repository to export.
+    required: true
+    type: str
+  product:
+    description:
+      - Name of the product that the repository belongs to.
+    required: true
+    type: str
+  chunk_size_gb:
+    description:
+      - Split the exported content into archives no greater than the specified size in gigabytes.
+    required: false
+    type: int
+  incremental:
+    description:
+      - Export only the content that has changed since the last export.
+    required: false
+    type: bool
+  from_history_id:
+    description:
+      - Export history identifier used for incremental export. If not provided the most recent export history will be used.
+    required: false
+    type: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.organization
+'''
+
+EXAMPLES = '''
+- name: "Export repository (full)"
+  content_export_repository:
+    id: 346
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+
+- name: "Export repository (full) in chunks of 10 GB"
+  content_export_repository:
+    id: 346
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    chunk_size_gb: 10
+
+- name: "Export repository (incremental) since the most recent export"
+  content_export_repository:
+      id: 346
+      username: "admin"
+      password: "changeme"
+      server_url: "https://foreman.example.com"
+      organization: "Default Organization"
+      incremental: true
+
+- name: "Export repository (incremental) since a specific export"
+  content_export_repository:
+      id: 346
+      username: "admin"
+      password: "changeme"
+      server_url: "https://foreman.example.com"
+      organization: "Default Organization"
+      incremental: true
+      from_history_id: 12345
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloAnsibleModule, _flatten_entity
+
+
+class KatelloContentExportModule(KatelloAnsibleModule):
+    pass
+
+
+def main():
+    module = KatelloContentExportModule(
+        foreman_spec=dict(
+            repository=dict(type='entity', flat_name='id', scope=['product']),
+            product=dict(type='entity', scope=['organization']),
+            chunk_size_gb=dict(required=False, type='int'),
+            from_history_id=dict(required=False, type='str'),
```

Comment:
I'd argue this should be an int and not a sting?
(And then also fix it in the other modules where we've overlooked that :see_no_evil: )

Reply:
yeah, int makes a lot more sense! I see your other PR now so I guess I'll leave that change over there.